### PR TITLE
feat(helm): introduce new installation and CRD lifecycle

### PR DESCRIPTION
**Pull request by oliverbaehler**
With this PR we are addressing the distribution of our CRDs and are trying to reconcile the Helm Chart as the main way of installing Capsule. With this change we can keep our overheads little and focus on one way of distribution. While implementing the CRDs i also did some refactoring of certain helm values. 

Implements:
-  #1063
- #998

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -29,6 +29,13 @@ builds:
           -X main.BuildTime={{ .Date }}
           -X main.GitRepo={{ .ProjectName }}
 release:
+  extra_files:
+    - glob: ./path/to/file.txt
```
Comment (maxgio92) [inline .goreleaser.yml:33]: Are these just sample files?

File: hack/argocd.yaml
Comment (maxgio92) [inline hack/argocd.yaml:1]: Is this just a sample? Because I don't think we want to ship these manifests. WDYT?

**Comment by MaxFedotov** [comment]
Hi @oliverbaehler,

what do you think about the method with a job that giantswarm uses to update crds -  https://github.com/giantswarm/kube-prometheus-stack-app/tree/a8315a8/helm/prometheus-operator-app/templates/crd-install. Seems like that will help to achieve what you want, but will be completely backward-compatible.

**Comment by prometherion** [comment]
+1 for @MaxFedotov approach, it's the same approach used by HAProxy Technologies for their Kubernetes Ingress Controller.

**Review by prometherion — APPROVED** [review]

**Review by maxgio92 — APPROVED** [review]
LGTM

**Pull request merged**